components/signoffs/signoffDao.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const { db, pgp, NotFoundError } = require('../../utils/dbUtils');
+const logger = require('../../utils/logger').logger;
+const queryStream = require('pg-query-stream');
+const qrec = pgp.errors.queryResultErrorCode;
+
+const id = 'id';
+const created = 'created';
+const caseIdentifier = 'case_identifier';
+const username = 'username';
+const qcPassed = 'qc_passed';
+const signoffStepName = 'signoff_step_name';
+const deliverableType = 'deliverable_type';
+
+const signoffCols = [
+  id,
+  created,
+  caseIdentifier,
+  username,
+  qcPassed,
+  signoffStepName,
+  deliverableType,
+];
+
+const signoffColsCreate = new pgp.helpers.ColumnSet(
+  [caseIdentifier, username, qcPassed, signoffStepName, deliverableType],
+  { table: 'signoff' }
+);
+
+const addSignoff = (signed) => {
+  return new Promise((resolve, reject) => {
+    db.task('add-signoffs', async (tx) => {
+      const signoffData = {
+        case_identifier: signed.caseIdentifier,
+        username: signed.username,
+        qc_passed: signed.qcPassed,
+        signoff_step_name: signed.signoffStepName,
+        deliverable_type: signed.deliverableType,
+      };
+
+      const signoffInsert = pgp.helpers.insert(
+        signoffData,
+        signoffColsCreate,
+        'signoff'
+      );
+      //add const onConflict here ?
+      const returning = ' RETURNING id';
+      const signoffQuery = signoffInsert + returning;
+    })
+      .then(() => {
```

Comment:
If you want to return the ID back to the controller, then you'll want to pass `data` as an argument to `then` - `.then((data)=> { resolve(data); })`. If your API contract doesn't require that, then no need for `RETURNING id`.